packages/core/src/baseclient.ts

```diff
@@ -282,9 +285,14 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
   /**
    * @inheritDoc
    */
-  public sendEvent(event: Event): void {
+  public sendEvent(event: Event, attachments?: Attachment[]): void {
```

Comment:
I wonder if we should expand this to take an entire `EventHint` to better future proof, and then we can keep the attachment in there.

Reply:
Yep good idea!

Reply:
I'm assuming that you're suggesting to add `attachments` to the existing `EventHint` type and use it in `sendEvent` too. 

That works and would also let you capture an event with attachments like without having to add it to the scope first:
```ts
captureEvent({
  ...someEvent
},
{
  attachments: [{ ...someAttachments}] 
});
``` 

But at what point would we copy the attachments from the `Scope` into that `hint`? In the capture/process/prepare/send pipeline, the scope on the hub only gets accessed in `_prepareEvent` but the hint is not returned. This comes back to the discussion earlier about allowing passing/mutating of the hint 🤔

Reply:
In python we add it to the hint: https://github.com/getsentry/sentry-python/blob/5eda9cf7f429f0aa67969062c93866827b0f282a/sentry_sdk/scope.py#L350

It seems like this gets mutated with `_prepare_event`: https://github.com/getsentry/sentry-python/blob/50ddda7b40c2d09b853b3fa2d595438c608a7eb0/sentry_sdk/client.py#L139 - and then it only returns an event.

Let's switch to allowing the hint to be mutated? It's kind of a hack - but it seems that the Java SDK is moving toward that: https://github.com/getsentry/sentry-javascript/issues/5036